ada/src/stemmer-factory.adb

```diff
@@ -0,0 +1,271 @@
+-----------------------------------------------------------------------
+--  stemmer-factory -- Multi-language stemmer with Snowball generator
+--  Copyright (C) 2020 Stephane Carrez
+--  Written by Stephane Carrez (Stephane.Carrez@gmail.com)
+--
+--  Licensed under the Apache License, Version 2.0 (the "License");
+--  you may not use this file except in compliance with the License.
+--  You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+--  Unless required by applicable law or agreed to in writing, software
+--  distributed under the License is distributed on an "AS IS" BASIS,
+--  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+--  See the License for the specific language governing permissions and
+--  limitations under the License.
```

Comment:
Would you be willing to license this the same way as the rest of Snowball (3 clause BSD)?

I'm not thrilled about the idea of introducing a mix of licenses - it's more confusing for users, and means developers have to take care when refactoring, etc.

Reply:
Not a problem. I'll change to 3 clause BSD for these files.

Reply:
Great, thanks.